src/process_utility.c

```diff
@@ -640,12 +647,47 @@ process_drop_hypertable_chunks(DropStmt *stmt)
 	return handled;
 }
 
+static void
+process_drop_hypertable_index(ProcessUtilityArgs *args, DropStmt *stmt)
```

Comment:
nit: I'd like a comment of why this is needed (pretty much what you said in the chat).